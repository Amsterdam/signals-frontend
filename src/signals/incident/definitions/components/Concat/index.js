import React from 'react';
import PropTypes from 'prop-types';
import isArray from 'lodash.isarray';

const Concat = ({ items }) => (
  <span>
    {items && items.map && items.map(item => {
      let key = item;
      if (React.isValidElement(item)) {
        if (item.props.items && isArray(item.props.items)) {
          key = item.props.items.join('-');
        } else if (item.props.children) {
          key = item.props.children;
        }
      }

      return <span key={key}>{item}</span>;
    })}
  </span>
);

Concat.propTypes = {
  items: PropTypes.arrayOf(PropTypes.shape).isRequired,
};

export default Concat;
