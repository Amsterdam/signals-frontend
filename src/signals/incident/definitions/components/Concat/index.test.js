/* eslint-disable react/prop-types */
import React from 'react';
import { shallow } from 'enzyme';

import Concat from './index';

describe('Definition component <Concat />', () => {
  const MockListComponent = ({ items }) => (
    <ul>
      {items.map(item => (
        <li key={item}>{item}</li>
      ))}
    </ul>
  );
  const MockChildComponent = ({ children }) => <span>{children}</span>;
  const MockComponent = () => <span>text 4</span>;
  let wrapper;

  describe('rendering', () => {
    it('should concat all texts and components of the array into spans', () => {
      wrapper = shallow(
        <Concat
          items={[
            'text 1',
            <MockListComponent items={['text 2']} />,
            <MockChildComponent>text 3</MockChildComponent>,
            <MockComponent />,
          ]}
        />
      );

      expect(wrapper).toMatchSnapshot();
    });
  });
});
