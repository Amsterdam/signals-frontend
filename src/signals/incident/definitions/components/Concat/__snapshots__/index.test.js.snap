// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Definition component <Concat /> rendering should concat all texts and components of the array into spans 1`] = `
<span>
  <span
    key="text 1"
  >
    text 1
  </span>
  <span
    key="text 2"
  >
    <MockListComponent
      items={
        Array [
          "text 2",
        ]
      }
    />
  </span>
  <span
    key="text 3"
  >
    <MockChildComponent>
      text 3
    </MockChildComponent>
  </span>
  <span
    key="[object Object]"
  >
    <MockComponent />
  </span>
</span>
`;
